File: api/sync.php
<?php
/**
 * API: /api/sync.php
 *
 * POST — синхронизация пользователей из ASU (MySQL → PostgreSQL)
 * Body: { "action": "create", "user": { id, name, login, password, email, phone } }
 */

header('Content-Type: application/json; charset=utf-8');

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    http_response_code(405);
    echo json_encode(['success' => false, 'message' => 'Method not allowed']);
    exit;
}

$body = json_decode(file_get_contents('php://input'), true);
$action = $body['action'] ?? null;
$user   = $body['user']   ?? null;

if ($action !== 'create' || !$user) {
    http_response_code(400);
    echo json_encode(['success' => false, 'message' => 'Invalid request']);
    exit;
}

// Разбиваем ФИО: "Фамилия Имя Отчество"
$parts     = explode(' ', trim($user['name'] ?? ''), 3);
$surname   = $parts[0] ?? '';
$firstname = $parts[1] ?? '';
$lastname  = $parts[2] ?? '';

try {
    $pdo = new PDO(
        'pgsql:host=localhost;port=5432;dbname=corporate_portal',
        'myuser',
        'VZAIMno4753',
        [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]
    );
    $pdo->exec("SET client_encoding = 'UTF8'");
} catch (PDOException $e) {
    http_response_code(500);
    echo json_encode(['success' => false, 'message' => 'DB connection error']);
    exit;
}

$stmt = $pdo->prepare("
    INSERT INTO public.user_info (id, status, login, password, firstname, surname, lastname, phone, email, ofo, user_group, role, auth)
    VALUES (:id, true, :login, :password, :firstname, :surname, :lastname, :phone, :email, -1, 'user', '', false)
    ON CONFLICT (id) DO NOTHING
");
$stmt->execute([
    ':id'        => (int)($user['id']),
    ':login'     => $user['login']    ?? '',
    ':password'  => $user['password'] ?? '',
    ':firstname' => $firstname,
    ':surname'   => $surname,
    ':lastname'  => $lastname,
    ':phone'     => $user['phone']    ?? '',
    ':email'     => $user['email']    ?? '',
]);

echo json_encode(['success' => true]);

File: backend/internal/handlers/sync.go
package handlers

import (
	"net/http"
	"strings"

	"github.com/jackc/pgx/v5/pgxpool"

	"corporate.admsr.ru/backend/internal/httpx"
)

type Sync struct {
	Pool *pgxpool.Pool
}

func (h *Sync) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		httpx.WriteJSON(w, http.StatusMethodNotAllowed, map[string]any{
			"success": false, "message": "Method not allowed",
		})
		return
	}
	var body struct {
		Action string `json:"action"`
		User   *struct {
			ID       any    `json:"id"`
			Name     string `json:"name"`
			Login    string `json:"login"`
			Password string `json:"password"`
			Email    string `json:"email"`
			Phone    string `json:"phone"`
		} `json:"user"`
	}
	if err := httpx.DecodeJSON(r, &body); err != nil || body.Action != "create" || body.User == nil {
		httpx.WriteJSON(w, http.StatusBadRequest, map[string]any{
			"success": false, "message": "Invalid request",
		})
		return
	}
	u := body.User
	parts := strings.SplitN(strings.TrimSpace(u.Name), " ", 3)
	surname, firstname, lastname := "", "", ""
	if len(parts) > 0 {
		surname = parts[0]
	}
	if len(parts) > 1 {
		firstname = parts[1]
	}
	if len(parts) > 2 {
		lastname = parts[2]
	}
	id := asInt64(u.ID)
	_, err := h.Pool.Exec(r.Context(), `
		INSERT INTO public.user_info
			(id, status, login, password, firstname, surname, lastname, phone, email, ofo, user_group, role, auth)
		VALUES ($1, true, $2, $3, $4, $5, $6, $7, $8, -1, 'user', '', false)
		ON CONFLICT (id) DO NOTHING`,
		id, u.Login, u.Password, firstname, surname, lastname, u.Phone, u.Email,
	)
	if err != nil {
		httpx.WriteJSON(w, http.StatusInternalServerError, map[string]any{
			"success": false, "message": "DB connection error",
		})
		return
	}
	httpx.WriteJSON(w, http.StatusOK, map[string]any{"success": true})
}
